request.go

```diff
@@ -18,13 +18,13 @@ func fillSearch(enc *encoder, spaceNo, indexNo uint32, key interface{}) error {
 	return enc.Encode(key)
 }
 
-func fillIterator(enc *encoder, offset, limit, iterator uint32) {
+func fillIterator(enc *encoder, offset, limit, iterator uint32) error {
 	encodeUint(enc, KeyIterator)
 	encodeUint(enc, uint64(iterator))
 	encodeUint(enc, KeyOffset)
 	encodeUint(enc, uint64(offset))
 	encodeUint(enc, KeyLimit)
-	encodeUint(enc, uint64(limit))
+	return encodeUint(enc, uint64(limit))
```

Comment:
Why it's the only field validated in `fillIteratior`?

Reply:
I have added error checking for all fields.